src/comm.cc

```diff
@@ -780,37 +774,43 @@ commCallCloseHandlers(int fd)
     }
 }
 
+/// sets SO_LINGER socket(7) option
+/// \param enabled -- whether linger will be active (sets linger::l_onoff)
+static void
+commConfigureLinger(const int fd, const OnOff enabled)
```

Comment:
The helper OnOff class enforces clients to pass an explicit enum parameter and prevents accidental mixing up integer and/or boolean parameters. The calls are prone to such confusion especially when there are several parameters, as we had at some previous branch commit: `commConfigureLinger(const int, const OnOff, const int timeoutInSeconds)`.
Since then we simplified, removing the last argument.

Reply:
> The calls are prone to such confusion especially when there are several parameters

Yes, this integer-like parameter mismatch wastes development/review time and influences "real bugs" like the one fixed by commit 38a9f558.

To better address the needs of callers like `eventAdd(long,list,of,integers,false)`, we may want to add more keywords to OnOff class or to add a few similar trivial enum classes, but this PR is a good starting point IMO.

Reply:
How does both a developer and an auditor mixup an FD parameter with a true/false value?

Reply:
> How does both a developer and an auditor mixup an FD parameter with a true/false value?

IMO, they can easily do that when they do not know which parameter comes first. This is especially true when there are more than two function parameters and/or default parameter values (e.g., see the fix in commit 38a9f558). The following code may look good to me as a reviewer, and I bet I have created similar bugs as a developer:

```C++
// TODO: Add cbdata-protection to LookupState()!
eventAdd("redo lookup a bit later if needed", &handler, Instance()->LookupState(true), 0, false);
```

Similar problems can happen when there are just two parameters as well. For example, the following function call may not raise red flags when audited by a reviewer (among some long/messy changes) or when accidentally modified (after copy-pasting correct code) by a new developer.

```C++
commConfigureLinger(!delayClosure, fd);
```


> The rational behind this seems a bit elitist.

FWIW, I see no connection between these changes rationale and "the belief that certain persons or members of certain groups deserve favored treatment by virtue of their superiority". I cannot even guess what group would be considered "elite" in this context; boolean parameters?!